### ISSUE TITLE
[BUG]: Application Fails to Exit PiP Mode When Clicking on App Icon

### ISSUE THREAD

**Issue by Sumeeth-24** [TheWidlarzGroup/react-native-video#3923]
### What happened?

When the application is in Picture-in-Picture (PiP) mode, clicking on the app icon does not restore the application to full screen. Instead, the application remains in PiP mode, displaying all the controls. The application only restores to full screen when the full-screen icon within the PiP window is clicked. However, clicking the application icon to restore it does not work as expected.

![pip-issue](https://github.com/TheWidlarzGroup/react-native-video/assets/61001158/3a10ad05-d44d-41e2-b85a-0524ff2af673)

Reproduction (screen recording): https://drive.google.com/file/d/1K4D-t8FFnWFe2izXxUIsJeNf8OHH0XuE/view?usp=sharing

I think the issue is related to `playInBackground` props.

**Expected Behavior:**

Clicking on the application icon should restore the application to full screen mode, similar to the behavior when the full-screen icon within the PiP window is clicked.

### Steps to reproduce

1. Launch the application
2. Enter PiP mode
3. Click on the app icon instead of the full-screen icon of the PiP window

### react-native-video version

6.2.0

### Platforms

iOS

### OS version

above 13

### Device

Real device, Simulator / Emulator

### Architecture

Old Architecture

**Comment by freeboub** [comment]
I thiink this should be something quiet easy to fix...
@YangJonghun did you find and fix this issue with your PIP PR ?

**Comment by ttnshubham** [comment]
@YangJonghun Please let us know if you find any solution. Replicating same issue at my end

**Comment by YangJonghun** [comment]
@freeboub
I haven't fix it

**Comment by YangJonghun** [comment]
@ttnshubham
Does this also happen to you when using PR #3385?

**Comment by ttnshubham** [comment]
@YangJonghun Yes, I am using same PR and updated with your latest changes too, I can replicate it.

**Comment by YangJonghun** [comment]
@ttnshubham 
Thanks for letting me know, I'll check it out :)

**Comment by ttnshubham** [comment]
@YangJonghun, have you got any time for this issue?

**Comment by YangJonghun** [comment]
@ttnshubham 
I found cause but I need more time to fix it.🥲

**Comment by kuldeepc0143** [comment]
how to implement PIP ?

**Comment by siddhant-rigi** [comment]
@YangJonghun Did you got time to solve it?

**Comment by TWG-AI** [comment]
Thanks for the screen recording — it clearly shows the expected vs. actual behavior. Tapping the app icon while in PiP should dismiss PiP and restore the app's full-screen presentation, which is standard iOS UX.

The root cause has been identified (the PiP delegate restoration path when the app is foregrounded via icon tap differs from the user tapping the expand button inside PiP). The fix will need to be applied in the PiP delegate implementation — specifically handling the case where the app is brought to the foreground without the user explicitly tapping the PiP expand button.
